File: src/mcp/embed_qes.ts
  import { CohereClient } from 'cohere-ai';

const cohere = new CohereClient({
  token: process.env.COHERE_API_KEY!,
});

export async function getQuestionEmbedding(question: string) {
  const response = await cohere.embed({
    texts: [question],
    model: 'embed-v4.0',
    inputType: 'search_query',
  });

  return response.embeddings[0];
}

File: src/mcp/embed_qes.py
import os
import cohere


def get_question_embedding(question: str) -> list[float]:
    """Get embedding for a question."""
    api_key = os.getenv("COHERE_API_KEY")
    co = cohere.Client(api_key=api_key)

    response = co.embed(
        texts=[question],
        model="embed-v4.0",
        input_type="search_query"
    )

    return response.embeddings[0]
